Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -211,17 +211,10 @@ public class DocumentIterator: IteratorProtocol {
     }
 
     /// Returns the current value. Assumes the iterator is in a valid position.
-    internal var currentValue: BSONValue? {
+    internal var currentValue: BSONValue {
         do {
-            switch self.currentType {
-            case .symbol:
-                return try Symbol.asString(from: self)
-            case .dbPointer:
-                return try DBPointer.asDocument(from: self)
-            default:
-                return try DocumentIterator.bsonTypeMap[currentType]?.init(from: self)
-            }
-        } catch {
+            return try self.safeCurrentValue()
```

Comment:
It looks like `safeCurrentValue()` already handled `.symbol` and `.dbPointer`. Was this refactoring unrelated to the PR itself?

Reply:
Yes, just removal of code duplication! 😄